bot/exts/events/advent_of_code/_cog.py

```diff
@@ -187,9 +187,10 @@ async def about_aoc(self, ctx: commands.Context) -> None:
     async def join_leaderboard(self, ctx: commands.Context) -> None:
         """DM the user the information for joining the Python Discord leaderboard."""
         current_date = datetime.now()
+        allowed_months = (Month.NOVEMBER.value, Month.DECEMBER.value)
         if (
```

Comment:
```suggestion
        if not (
```
Since, the first batch of condition will result in False, and the second one as well, since it's January and `2021+1`

Reply:
Why should the bot say the leaderboard isn't available if it's the right month?

Reply:
I'm not sure I follow, this `if` block is to determine if it's not current AoC, if it's not, then we send the error and return.

Current AoC is defined by the AOC_YEAR var, which we set in the env. So currently, it's set to 2021.

Both parts would fail right now, so the user invoking can join the 2021 board. Come Feb, neither of the conditionals will pass, so the error comes up and the command returns early.

Reply:
I tested it out locally, and it did not work.
Must have overlooked something.

Reply:
Did you set the AOC_YEAR env var to 2021 locally?

Reply:
Yes, of course.
Event tested out with 2022 just to make sure it won't work. (And it will with 2021)

Reply:
No wait, you are right, sorry. i need to look into this more.

Reply:
Alright, pushed a fix.

Reply:
Oh, alright.
The logic checked out but something was off 👍

Reply:
Yea, too much negation, ands and ors combined. This should read a little easier now.